CMakeLists.txt

```diff
@@ -1,27 +1,41 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(pcl_conversions)
 
-find_package(catkin REQUIRED COMPONENTS pcl_msgs roscpp sensor_msgs std_msgs cmake_modules)
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra")
+endif()
 
-find_package(PCL REQUIRED QUIET COMPONENTS common)
-find_package(Eigen REQUIRED)
+find_package(ament_cmake REQUIRED)
 
-include_directories(include ${catkin_INCLUDE_DIRS} ${PCL_COMMON_INCLUDE_DIRS} ${Eigen_INCLUDE_DIRS})
+find_package(Boost REQUIRED)
```

Comment:
If you find_package Boost you need to add it to the dependencies in the package.xml.

In this case you don't use boost anywhere (the only occurences of boost are for shared_ptr and commented out) so I'd suggest removing the find_package and convert them to std::shared_ptr